lib/_http_parser.js

```diff
@@ -0,0 +1,890 @@
+/*
```

Comment:
> should be it's own library and vendored in like readable-stream is

readable-stream still pulls from iojs itself, because the library has different goals than iojs itself (it favors broad compatibility where io.js favors performance.) It seems to me like the JS parser is likely to make similar tradeoffs and thus run into the same problems with vendoring.